selene-lib/src/rules.rs

```diff
@@ -55,8 +55,8 @@ pub trait Rule {
         ast_context: &AstContext,
     ) -> Vec<Diagnostic>;
 
-    fn severity(&self) -> Severity;
-    fn rule_type(&self) -> RuleType;
+    fn severity() -> Severity;
```

Comment:
Any chance you can just make these `const SEVERITY: Severity` etc? If not these should be const fn

Reply:
Sure! That'll be an annoying regex to write though